ports/stm32/adc.c

```diff
@@ -446,9 +458,18 @@ STATIC void adc_config_channel(ADC_HandleTypeDef *adc_handle, uint32_t channel)
 }
 
 STATIC uint32_t adc_read_channel(ADC_HandleTypeDef *adcHandle) {
-    HAL_ADC_Start(adcHandle);
-    adc_wait_for_eoc_or_timeout(adcHandle, EOC_TIMEOUT);
-    uint32_t value = adcHandle->Instance->DR;
+    uint32_t value;
+    #if defined(STM32G4)
+    // For STM32G4, there is a errata on ADC that will get wrong ADC result.
+    // According to the errata sheet, this can be avoid by performing
+    // two consecutive ADC conversions and keep second result.
```

Comment:
Can you please point to the exact errata for this?  (PDF document and errata number.)

Reply:
https://www.st.com/resource/en/errata_sheet/es0430-stm32g471xx473xx474xx483xx484xx-device-errata-stmicroelectronics.pdf
2.7.7 Wrong ADC result if conversion done late after calibration or previous conversion
![stm32g474re-errata](https://github.com/micropython/micropython/assets/47820079/951584aa-02d9-49a2-a57f-000042981e85)

Without this workaround, the adc result is lower than actual one. It may be affected by this errata because it takes about 1~2ms between `HAL_ADCEx_Calibration_Start()` and `HAL_ADC_Start()` on `adc.c`.

Reply:
OK, thanks for the reference.  I have updated the comment in the code to refer to this 2.7.7 errata.